example/m5_competition.py

```diff
@@ -0,0 +1,311 @@
+was then used to train a collection of Gradient Boosted Trees models.
+
+See https://www.kaggle.com/competitions/m5-forecasting-accuracy/overview.
+
+In this example, we will explore how to use the Temporian library to replicate
+this transformation from time series into a tabular dataset. By following the
+step-by-step guide provided, you will learn how to leverage the library's
+powerful functionalities to replicate transformations of your time series data.
+With Temporian, converting time series data into a tabular format has never been
+easier.
+"""
+
+import os
+import temporian as tp
+import pandas as pd
+import numpy as np
+from datetime import datetime, timezone, timedelta
+import urllib.request
+import zipfile
+import tempfile
+
+# Directory used to download the raw M5 dataset and to export the tabular
+# dataset.
+work_directory = os.path.join(tempfile.gettempdir(), "m5")
+os.makedirs(work_directory, exist_ok=True)
+print("Use work directory:", work_directory)
+
+# Download the M5 dataset
+raw_data_zip = os.path.join(work_directory, "raw.zip")
+if not os.path.exists(raw_data_zip):
+    print("Download M5 dataset in", raw_data_zip)
+    url = "https://docs.google.com/uc?export=download&id=1NYHXmgrcXg50zR4CVWPPntHx9vvU5jbM&confirm=t"
+    urllib.request.urlretrieve(url, raw_data_zip)
+
+# Extract the M5 dataset
+raw_data_dir = os.path.join(work_directory, "raw")
+if not os.path.exists(raw_data_dir):
+    print("Extract M5 dataset in", raw_data_dir)
+    with zipfile.ZipFile(raw_data_zip, "r") as zip_ref:
+        zip_ref.extractall(raw_data_dir)
+
+
+# Load raw dataset
+print("Load raw dataset")
+print("================")
+
+# During development, set nrows to a small value (e.g. nrows=100) to
+# only load a sample of the dataset.
+#
+# TODO: Set to None when fast enough.
+nrows = 100
+
+raw_path = lambda x: os.path.join(raw_data_dir, x)
+sales_raw = pd.read_csv(raw_path("sales_train_evaluation.csv"), nrows=nrows)
+sell_prices_raw = pd.read_csv(raw_path("sell_prices.csv"), nrows=nrows)
+calendar_raw = pd.read_csv(raw_path("calendar.csv"))
+
+print("sales_raw:")
+sales_raw.info()
+print("sell_prices_raw:")
+sell_prices_raw.info()
+print("calendar_raw:")
+calendar_raw.info()
+
+# Melt dataset
+#
+# The different timesteps of sale data ("sales_raw") are stored in different
+# columns. Instead, we want for each timesteps to be a different row.
+
+print("Melt dataset")
+print("============")
+
+sales_raw = pd.melt(
+    sales_raw,
+    var_name="day",
+    value_name="sales",
+    id_vars=["id", "item_id", "dept_id", "cat_id", "store_id", "state_id"],
+)
+sales_raw["day"] = sales_raw["day"].apply(lambda x: int(x[2:]))
+
+print("sales_raw:\n", sales_raw)
+
+# Compute timestamps
+#
+# In the M5 raw dataset, timestamps are stored in three different ways in the
+# three different tables. To make our life easy, we will convert all the
+# timestamps into python datetimes objects.
+
+print("Compute timestamps")
+print("==================")

```

Comment:
I'd add to some or all of these lines line breaks: `print("\n\nCompute timestamps")`, the output has many lines and I found it a bit hard to parse.

Reply:
Meta-comment: The objective of this file is for us to test and improve our API. This is not a tutorial, and it will have to be re-worked before being something for the users. All the raised points are good, and I keep them for this re-work.